File: Printpress.Application/Reports/DTO/InventoryPurchaseReportDto.cs
namespace Printpress.Application;

public class InventoryPurchaseLineRowDto
{
    public Guid InvoiceId { get; set; }
    public string InvoiceNumber { get; set; }
    public DateTime InvoiceDate { get; set; }
    public string SupplierName { get; set; }
    public Guid ItemId { get; set; }
    public string ItemName { get; set; }
    public string CategoryName { get; set; }
    public decimal Quantity { get; set; }
    public decimal UnitPrice { get; set; }
    public decimal LineTotal { get; set; }
}

public class InventoryPurchaseReportDto
{
    public List<InventoryPurchaseLineRowDto> Lines { get; set; } = [];
    public int InvoiceCount { get; set; }
    public int LineCount { get; set; }
    public decimal TotalQuantity { get; set; }
    public decimal TotalAmount { get; set; }
}

File: Printpress.UI/Printpress_Angular/src/app/features/reports/models/inventory-purchase-report.dto.ts
export interface InventoryPurchaseLineRowDto {
  invoiceId: string;
  invoiceNumber: string;
  invoiceDate: string;
  supplierName: string;
  itemId: string;
  itemName: string;
  categoryName: string;
  quantity: number;
  unitPrice: number;
  lineTotal: number;
}

export interface InventoryPurchaseReportDto {
  lines: InventoryPurchaseLineRowDto[];
  invoiceCount: number;
  lineCount: number;
  totalQuantity: number;
  totalAmount: number;
}
